modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParser.java

```diff
@@ -399,6 +402,11 @@ public ExtensionDevelopmentFramework getDevelopmentFramework() {
     return JAVA_SDK;
   }
 
+  @Override
+  public Optional<ArtifactLifecycleListener> getArtifactLifecycleListener() {
+    return new ArtifactLifecycleListenerParser().getArtifactLifecycleListener(extensionElement);
```

Comment:
Just thought it would help to avoid polluting this Parser with too many things. But now that we added the Factory as its own entity it makes even less sense to have a dedicated Parser for the Listener.
So I just removed it and moved its two private methods here.